File: services/auth-service/src/Auth/AuthErrorCodes.cs
namespace AuthService.Auth;

/// <summary>
/// Codici stabili degli errori auth. I client li usano per decidere comportamento
/// (es. mostrare il pulsante "reinvia email") e per la localizzazione dei messaggi.
/// I valori non vanno mai cambiati: aggiungerne di nuovi è ok.
/// </summary>
public static class AuthErrorCodes
{
    // 401 — login / token
    public const string InvalidCredentials       = "invalid_credentials";
    public const string EmailNotConfirmed        = "email_not_confirmed";
    public const string InvalidConfirmationToken = "invalid_confirmation_token";
    public const string UserNotFound             = "user_not_found";

    // 401 — Google
    public const string InvalidGoogleToken       = "invalid_google_token";
    public const string GoogleEmailNotVerified   = "google_email_not_verified";
    public const string GoogleAccountCreateFailed = "google_account_create_failed";

    // 409 — registrazione
    public const string EmailAlreadyRegistered   = "email_already_registered";
    public const string UsernameAlreadyTaken     = "username_already_taken";
    public const string PasswordRejected         = "password_rejected";
}

File: frontend/src/services/AuthErrorCodes.js
// Codici stabili degli errori auth (devono restare in sync con il backend).
// Per l'i18n: sostituisci AUTH_ERROR_MESSAGES con una funzione t(code) dell'i18n.
export const AuthErrorCodes = {
  InvalidCredentials:        'invalid_credentials',
  EmailNotConfirmed:         'email_not_confirmed',
  InvalidConfirmationToken:  'invalid_confirmation_token',
  UserNotFound:              'user_not_found',
  InvalidGoogleToken:        'invalid_google_token',
  GoogleEmailNotVerified:    'google_email_not_verified',
  GoogleAccountCreateFailed: 'google_account_create_failed',
  EmailAlreadyRegistered:    'email_already_registered',
  UsernameAlreadyTaken:      'username_already_taken',
  PasswordRejected:          'password_rejected',
};

// Dizionario IT — qui andrebbe la chiamata a un sistema i18n in futuro.
const AUTH_ERROR_MESSAGES_IT = {
  [AuthErrorCodes.InvalidCredentials]:        'Email o password non corretti.',
  [AuthErrorCodes.EmailNotConfirmed]:         'Email non confermata. Controlla la tua casella di posta.',
  [AuthErrorCodes.InvalidConfirmationToken]:  'Link di conferma non valido o scaduto.',
  [AuthErrorCodes.UserNotFound]:              'Utente non trovato.',
  [AuthErrorCodes.InvalidGoogleToken]:        'Token Google non valido.',
  [AuthErrorCodes.GoogleEmailNotVerified]:    'La tua email Google non è verificata.',
  [AuthErrorCodes.GoogleAccountCreateFailed]: 'Impossibile creare l\'account con Google.',
  [AuthErrorCodes.EmailAlreadyRegistered]:    'Questa email è già registrata.',
  [AuthErrorCodes.UsernameAlreadyTaken]:      'Username già in uso. Scegline un altro.',
  [AuthErrorCodes.PasswordRejected]:          'La password non rispetta i requisiti.',
};

/**
 * Risolve un codice d'errore auth in un messaggio user-facing.
 * Fallback: usa il messaggio del server, poi un generico.
 */
export const authMessage = (code, serverMessage) =>
  AUTH_ERROR_MESSAGES_IT[code] || serverMessage || 'Si è verificato un errore.';
